.github/workflows/check-new-library-versions.yml

```diff
@@ -0,0 +1,142 @@
+name: "Check new library versions"
+
+on:
+  schedule:
+    - cron: "0 0 8 * *"
+    - cron: "0 0 22 * *"
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  actions: write
+
+concurrency:
+  group: "workflow = ${{ github.workflow }}, ref = ${{ github.event.ref }}, pr = ${{ github.event.pull_request.id }}"
+  cancel-in-progress: true
+
+jobs:
+  get-all-libraries:
+    if: github.repository == 'oracle/graalvm-reachability-metadata'
+    name: "📋 Get list of all supported libraries with newer versions"
+    permissions: write-all
+    runs-on: "ubuntu-20.04"
+    timeout-minutes: 5
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      issue: ${{ steps.set-issue.outputs.issue }}
+    steps:
+      - name: "☁️ Checkout repository"
+        uses: actions/checkout@v4
+      - name: "🔧 Prepare environment"
+        uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '21'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew getExistingLibrariesWithNewVersions -PlibraryMatrixLimit=200
+      - name: "🔨 Create branch"
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "Github Actions"
+          git switch -C libraryVersionsDependabot/$(date '+%Y-%m-%d')
+          git push origin libraryVersionsDependabot/$(date '+%Y-%m-%d')
```

Comment:
I see you are using such branch names yourself, but typically, we use kebab-case instead of camelCase. Also, why do you reference dependabot? Does GH's dependabot really do anything here? I guess using the date, also only really works because the jobs are scheduled at midnight. Otherwise, it's possible that the branch is created on a different date before it is checked out by the next job(s).